ci/.gitlab-ci.integration.yml

```diff
@@ -5,17 +5,17 @@ variables:
 
 setup:integration-image:
   image: docker:19.03.12
-  stage: .pre
+  stage: build
```

Comment:
why move it to build, if i may ask?

Reply:
This new image depends on the base one (i.e. `dev/base.Dockerfile`) It has to wait for it to finish before building this one. 
Since this image will be used in the integration job during testing, this felt like the logical place to put it. But to be honest, coming back to this today, I makes even more sense to build this image in the same step as the first one (although, that has a cost in terms of pipeline duration, but at least we leverage docker caching 🤷🏻‍♂️)

Reply:
so... can we not move it to the same stage and than create a dependency between the two jobs?

Reply:
Yeah, let me try that

Reply:
We can create dependencies between jobs in the same stage. I either keep them in the same job or leave the second on during the build stage. 
Considering that `only: changes` doesn't work with our setup, they will always both run. So keeping them separated would be only for logical purposes.